Sources/Networking/Operations/GetIntroEligibilityOperation.swift

```diff
@@ -86,40 +74,28 @@ private extension GetIntroEligibilityOperation {
                                                      fetchToken: self.receiptData.asFetchToken)),
                                   path: .getIntroEligibility(appUserID: appUserID))
 
-        httpClient.perform(request, authHeaders: self.authHeaders) { (response: HTTPResponse<[String: Any]>.Result) in
-            let eligibilityResponse = IntroEligibilityResponse(result: response.mapError { $0.asPurchasesError },
-                                                               productIdentifiers: self.productIdentifiers,
-                                                               unknownEligibilityClosure: unknownEligibilityClosure,
-                                                               completion: self.responseHandler)
-            self.handleIntroEligibility(response: eligibilityResponse)
+        httpClient.perform(
+            request, authHeaders: self.authHeaders
+        ) { (response: HTTPResponse<GetIntroEligibilityResponse>.Result) in
+            self.handleIntroEligibility(result: response,
+                                        productIdentifiers: self.productIdentifiers,
+                                        completion: self.responseHandler)
             completion()
         }
     }
 
-    func handleIntroEligibility(response: IntroEligibilityResponse) {
-        let result: [String: IntroEligibility] = {
-            var eligibilitiesByProductIdentifier = response.result.value?.body
+    func handleIntroEligibility(
+        result: Result<HTTPResponse<GetIntroEligibilityResponse>, NetworkError>,
+        productIdentifiers: [String],
+        completion: Backend.IntroEligibilityResponseHandler
+    ) {
+        let eligibilities = result.value?.body.eligibilityByProductIdentifier
 
-            if response.result.value?.statusCode.isSuccessfulResponse != true {
-                eligibilitiesByProductIdentifier = [:]
-            }
-
-            guard let eligibilitiesByProductIdentifier = eligibilitiesByProductIdentifier else {
-                return response.unknownEligibilityClosure()
-            }
+        let result: [String: IntroEligibility] = Set(productIdentifiers)
+            .dictionaryWithValues { eligibilities?[$0] ?? .unknown }
```

Comment:
total nitpick, feel free to ignore: 
at first glance I thought this was `eligibilities[0]` and thought that was weird, then I realized I had missed the `$`. 
I think for places where we're reading using `[ ]` we could name the variable for the sake of readability
```suggestion
            .dictionaryWithValues { productID in eligibilities?[productID] ?? .unknown }
```

Reply:
Yup that's a good point.